stacks-signer/src/signerdb.rs

```diff
@@ -70,6 +83,41 @@ impl SignerDb {
         )
     }
 
+    /// Get the signer state for the provided reward cycle if it exists in the database
+    pub fn get_signer_state(&self, reward_cycle: u64) -> Result<Option<SignerState>, DBError> {
+        let result: Option<String> = query_row(
+            &self.db,
+            "SELECT state FROM signer_states WHERE reward_cycle = ?",
+            &[u64_to_sql(reward_cycle)?],
+        )?;
+
+        try_deserialize(result)
+    }
+
+    /// Insert the given state in the `signer_states` table for the given reward cycle
+    pub fn insert_signer_state(
+        &self,
+        reward_cycle: u64,
+        signer_state: &SignerState,
+    ) -> Result<(), DBError> {
+        let serialized_state = serde_json::to_string(signer_state)?;
+        self.db.execute(
+            "INSERT OR REPLACE INTO signer_states (reward_cycle, state) VALUES (?1, ?2)",
+            &[&u64_to_sql(reward_cycle)? as &dyn ToSql, &serialized_state],
+        )?;
+        Ok(())
+    }
+
+    /// Delete the signer state for the provided reward cycle and signer ID
+    pub fn delete_signer_state(&self, reward_cycle: u64) -> Result<(), DBError> {
```

Comment:
It's unused. I added it to just for the sake of completeness of the interface, and to be consistent with the `insert_block`, `remove_block` functions. `remove_block` is also unused. If you prefer to call YAGNI here, I can happily remove them. Let me know what you think.

Reply:
Actually, since there are other considerations that are more relating to this module than the current PR - I'd rather agree on what we want to do and create a separate PR for that refactor. Topics for this would be:

1. Return results or do eager panics?
2. `rusqlite::params` vs `&dyn ToSql`
3. Prune unused removal functions?

Reply:
1. Return results
2. `rusqlite::params`
3. YAGNI
4. Do it all in this PR.  We're talking less than 10 lines of refactoring.

Reply:
Done.